Zotero/Scenes/Detail/PDF/ViewModels/PDFReaderActionHandler.swift

```diff
@@ -2135,8 +2136,17 @@ final class PDFReaderActionHandler: ViewModelActionHandler, BackgroundDbProcessi
 
         DDLogInfo("PDFReaderActionHandler: database annotation changed")
 
-        // Get sorted database keys
-        var keys = (viewModel.state.snapshotKeys ?? viewModel.state.sortedKeys).filter({ $0.type == .database })
+        // Keys for deletions and modifications are indexed by the previously observed items
+        var previousObservedKeys: [PDFReaderState.AnnotationKey] = []
```

Comment:
Can we just directly use `viewModel.state.databaseAnnotations`? We do that on line 2225 anyway. No real need to extract keys separately.

Reply:
Good point (I planned to improve that, after changing font size to use a float, but better if I do it in a separate PR)
Check https://github.com/zotero/zotero-ios/pull/999/commits/6e0b28fbb1a3b464b94c6cc4bbe698c5c366c8cb